core/src/Kokkos_Serial.hpp

```diff
@@ -573,13 +570,14 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
             : pointer_type(
                   internal_instance->m_thread_team_data.pool_reduce_local());
 
-    reference_type update =
-        ValueInit::init(ReducerConditional::select(m_functor, m_reducer), ptr);
+    typename Analysis::Reducer final_reducer(
+        &ReducerConditional::select(m_functor, m_reducer));
+
+    reference_type update = final_reducer.init(ptr);
```

Comment:
Why do you call it "final" reducer?

Reply:
Good question! For me, it's the final reducer to be used after wrapping the given Functor and Reducer also handling `init`, `join`, and `final`. That being said, I'm happy to take other suggestions. Currently, it's used everywhere in #4663.